src/librustc_typeck/check/method/probe.rs

```diff
@@ -1033,7 +1117,15 @@ impl<'a, 'gcx, 'tcx> ProbeContext<'a, 'gcx, 'tcx> {
             return Some(Err(MethodError::Ambiguity(sources)));
         }
 
-        applicable_candidates.pop().map(|probe| Ok(probe.to_unadjusted_pick()))
+        let ret: Vec<_> = applicable_candidates.iter()
+                                               .map(|probe| probe.to_unadjusted_pick())
+                                               .collect();
+
+        if ret.len() < 1 {
```

Comment:
Now that `PickResult` is already a `Result<Vec<->, ->`, I don't think we need that `Option`.